Frontend/Bookings/Bookings.jsx

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import { useContext, useState, useEffect } from "react";
+import AuthContext from "../context/AuthProvider";
+import { getAllTrainings, getAllBookings } from "../../Utils/DataManagement";
+import { ChevronLeftIcon, ChevronRightIcon } from "@heroicons/react/solid";
+import {
+  eachDayOfInterval,
+  format,
+  startOfWeek,
+  startOfToday,
+  endOfMonth,
+  endOfWeek,
+  add,
+  parse,
+  isEqual,
+  isSameMonth,
+  isToday,
+  isSameDay,
+  parseISO,
+} from "date-fns";
+import ViewTraineeBookings from "../Trainee/ViewTraineeBookings/ViewTraineeBookings";
+import TrainerBookedSessions from "../Trainer/TrainerBookedSessions/TrainerBookedSessions";
+import NavBar from "../NavBar/NavBar";
+
+import "./Bookings.css";
+
+export default function FindTraining() {
+  const { chooseClass, userName } = useContext(AuthContext);
+  const [match, setMatch] = useState([]);
+  const [found, setFound] = useState([]);
+
+  let today = startOfToday();
+  let [selectedDay, setSelectedDay] = useState(today);
+  let [currentMonth, setCurrentMonth] = useState(format(today, "MMM-yyyy"));
+  let firstDayCurrentMonth = parse(currentMonth, "MMM-yyyy", new Date());
+  let days = eachDayOfInterval({
+    start: startOfWeek(firstDayCurrentMonth),
+    end: endOfWeek(endOfMonth(firstDayCurrentMonth)),
+  });
+  function nextMonth() {
+    let firstDayCurrentMonth = parse(currentMonth, "MMM-yyyy", new Date());
+    let firstDayNextMonth = add(firstDayCurrentMonth, { months: 1 });
+    setCurrentMonth(format(firstDayNextMonth, "MMM-yyyy"));
+  }
+  function previousMonth() {
+    let firstDayCurrentMonth = parse(currentMonth, "MMM-yyyy", new Date());
+    let firstDayNextMonth = add(firstDayCurrentMonth, { months: -1 });
+    setCurrentMonth(format(firstDayNextMonth, "MMM-yyyy"));
+  }
+  useEffect(() => {
+    if (chooseClass === "Trainer") {
+      async function Training() {
+        setMatch(await getAllTrainings());
+        setFound(
+          match?.filter((each) => {
+            return each.fullname === userName;
+          })
+        );
+      }
+      Training();
+    }
+  }, [match]);
+
+  useEffect(() => {
+    if (chooseClass === "Trainee") {
+      async function Booking() {
+        setMatch(await getAllBookings());
+        setFound(
+          match.data.filter((each) => {
+            return each.traineeName === userName;
+          })
+        );
+      }
+      Booking();
+    }
+  }, [match]);
+
+  function classNames(...classes) {
+    return classes.filter(Boolean).join(" ");
+  }
+  return (
+    <>
+      {chooseClass === "Trainee" ? <NavBar /> : null}
+      <div className="pt-16">
+        <div className="max-w-md px-4 mx-auto sm:px-7 md:max-w-4xl md:px-6">
+          <div className="md:grid md:grid-cols-2 md:divide-x md:divide-gray-200">
+            <div className="md:pr-14">
```

Comment:
I think the Calendar can be its own control. This file has gotten a little too big. see you can split this further.